src/providers/tfc/planProvider.ts

```diff
@@ -0,0 +1,424 @@
+    }
+    if (planLog && planLog.driftChanges) {
+      items.push(new DriftChangesItem(planLog.driftChanges, planLog.driftSummary));
+    }
+    //  && planLog.outputs.size > 0
+    if (planLog && planLog.outputs) {
+      items.push(new OutputsItem(planLog.outputs));
+    }
+    if (planLog && planLog.diagnostics && planLog.diagnosticSummary && planLog.diagnostics.length > 0) {
+      items.push(new DiagnosticsItem(planLog.diagnostics, planLog.diagnosticSummary));
+    }
+    return items;
+  }
+
+  private async getPlanFromUrl(plan: PlanTreeItem): Promise<PlanLog | undefined> {
+    const session = await vscode.authentication.getSession(TerraformCloudAuthenticationProvider.providerID, [], {
+      createIfNone: false,
+    });
+
+    if (session === undefined) {
+      return;
+    }
+
+    try {
+      const result = await axios.get(plan.logReadUrl, {
+        headers: { Accept: 'text/plain' },
+        responseType: 'stream',
+      });
+      const lineStream = readline.createInterface({
+        input: result.data,
+        output: new Writable(),
+      });
+
+      const planLog: PlanLog = {};
+
+      for await (const line of lineStream) {
+        try {
+          const logLine: LogLine = JSON.parse(line);
+
+          if (logLine.type === 'planned_change' && logLine.change) {
+            if (!planLog.plannedChanges) {
+              planLog.plannedChanges = [];
+            }
+            planLog.plannedChanges.push(logLine.change);
+            continue;
+          }
+          if (logLine.type === 'resource_drift' && logLine.change) {
+            if (!planLog.driftChanges) {
+              planLog.driftChanges = [];
+            }
+            if (!planLog.driftSummary) {
+              planLog.driftSummary = {
+                changed: 0,
+                deleted: 0,
+              };
+            }
+            planLog.driftChanges.push(logLine.change);
+            if (logLine.change.action === 'update') {
+              planLog.driftSummary.changed += 1;
+            }
+            if (logLine.change.action === 'delete') {
+              planLog.driftSummary.deleted += 1;
+            }
+            continue;
+          }
+          if (logLine.type === 'change_summary' && logLine.changes) {
+            planLog.changeSummary = logLine.changes;
+            continue;
+          }
+          if (logLine.type === 'outputs' && logLine.outputs) {
+            // JSON.parse() won't do the conversion from [JSON] object
+            // to the Map so we have to do it here.
+            Object.entries(logLine.outputs).forEach(function (value: [string, OutputChange]) {
+              if (!planLog.outputs) {
+                planLog.outputs = new Map();
+              }
+              planLog.outputs.set(value[0], value[1]);
+            });
+
+            continue;
+          }
+          if (logLine.type === 'diagnostic' && logLine.diagnostic) {
+            if (!planLog.diagnostics || !planLog.diagnosticSummary) {
```

Comment:
```suggestion
            if (!planLog.diagnostics && !planLog.diagnosticSummary) {
```
Shouldn't this be an and, so we don't accidentally overwrite data?